### PR TITLE
perl: allow building with intel@19.1.3

### LINKED ISSUE #25418: Installation issue: perl
### Steps to reproduce the issue

spack install perl %intel@19.1.3.304
This is with the latest pull of spack and installs perl@5.34. Similar errors observed with 5.32.1 and 5.35.0. 5.32.0 works. (EDIT)

Numerous errors occur in the build phase after a successful configure phase starting with:
  >> 38      gcc: error: unrecognized command line option '-we147'
  >> 39      gcc: error: unrecognized command line option '-mp'; did you mean '-p'?
  >> 40      gcc: error: unrecognized command line option '-no-gcc'; did you mean '-fno-dce'?

and followed by dozens similar to:
  >> 959     inline.h(149): error: identifier "__INT_MAX__" is undefined
     960           if (COP_SEQ_RANGE_LOW(pn) == PERL_PADSEQ_INTRO)
     961                                        ^
     962     
  >> 963     pad.c(329): error: type name is not allowed
     964                   if (CvSLABBED(&cvbody)) OpslabREFCNT_dec_padok(OpSLAB(CvROOT(&cvbody)));

### Information on your system

This occurs building on NERSC's CORI system.

* **Spack:** 0.16.2-3901-264b00bff4
* **Python:** 3.8.5
* **Platform:** cray-sles15-haswell
* **Concretizer:** original

### Additional information

@mjwoods @hartzell @justintoo (There are Author credits in the package but no maintainers tagged. Maybe worth updating if these devs are still active?)

[spack-build-env.txt](https://github.com/spack/spack/files/6983836/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/6983837/spack-build-out.txt)
[spack-configure-args.txt](https://github.com/spack/spack/files/6983838/spack-configure-args.txt)

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
For some strange reason, intel 19.1.3.x will not build perl (see https://github.com/spack/spack/issues/25418). This PR adds a patch that gets run for `perl %intel@19.1.3` and removes some nonexistent flags from `hints/linux.sh` that for some reason create trouble in the build.

Fixes https://github.com/spack/spack/issues/25418

### CODE REVIEW

File: var/spack/repos/builtin/packages/perl/package.py [hunk 1]
```diff
@@ -154,6 +154,10 @@ def patch(self):
         # https://github.com/Perl/perl5/issues/15544 long PATH(>1000 chars) fails a test
         os.chmod("lib/perlbug.t", 0o644)
         filter_file("!/$B/", "! (/(?:$B|PATH)/)", "lib/perlbug.t")
+        # Several non-existent flags cause Intel@19.1.3.304 to fail
+        with when("%intel@19.1.3.304"):
```
Comment (RemiLacroix-IDRIS) [inline var/spack/repos/builtin/packages/perl/package.py]: Maybe the condition on the version should be relaxed a bit, we use 19.1.3 here for example.

**Review by RemiLacroix-IDRIS — COMMENTED** [review]
I have had this issue for ages, never bothered to try fixing it.

Note that version 5.32.0 (and probably older versions) do work with Intel 19.1.3.

File: var/spack/repos/builtin/packages/perl/package.py (hunk 1 above)
Comment (AlexanderRichert-NOAA) [reply var/spack/repos/builtin/packages/perl/package.py]: Makes sense. Just changed it to 19.1.3.

**Comment by AlexanderRichert-NOAA** [comment]
@spackbot re-run pipeline

**Comment by AlexanderRichert-NOAA** [comment]
@spackbot re-run pipeline

**Comment by AlexanderRichert-NOAA** [comment]
@spackbot re-run pipeline

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
@spackbot run pipeline

**Review by alalazo — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Comment by AlexanderRichert-NOAA** [comment]
@spackbot run pipeline

**Comment by scottwittenburg** [comment]
I was investigating this branch because pipeline generation appeared not to be pruning dependents beyond level 1 parents.  Then I noticed it just needs to run merged with a newer develop (one that includes #35944).  Asking spackbot to run a pipeline doesn't create a new merge commit, so I updated the branch.  Let's see what happens with the next pipeline.

**Pull request merged**